pkg/scheduler/scheduler.go

```diff
@@ -305,45 +307,78 @@ func New(client clientset.Interface,
 		}
 		options.profiles = cfg.Profiles
 	}
-	schedulerCache := internalcache.New(durationToExpireAssumedPod, stopEverything)
 
 	registry := frameworkplugins.NewInTreeRegistry()
 	if err := registry.Merge(options.frameworkOutOfTreeRegistry); err != nil {
 		return nil, err
 	}
 
-	snapshot := internalcache.NewEmptySnapshot()
-	clusterEventMap := make(map[framework.ClusterEvent]sets.String)
+	metrics.Register()
 
-	configurator := &Configurator{
-		componentConfigVersion:       options.componentConfigVersion,
-		client:                       client,
-		kubeConfig:                   options.kubeConfig,
-		recorderFactory:              recorderFactory,
-		informerFactory:              informerFactory,
-		schedulerCache:               schedulerCache,
-		StopEverything:               stopEverything,
-		percentageOfNodesToScore:     options.percentageOfNodesToScore,
-		podInitialBackoffSeconds:     options.podInitialBackoffSeconds,
-		podMaxBackoffSeconds:         options.podMaxBackoffSeconds,
-		podMaxUnschedulableQDuration: options.podMaxUnschedulableQDuration,
-		profiles:                     append([]schedulerapi.KubeSchedulerProfile(nil), options.profiles...),
-		registry:                     registry,
-		nodeInfoSnapshot:             snapshot,
-		extenders:                    options.extenders,
-		frameworkCapturer:            options.frameworkCapturer,
-		parallellism:                 options.parallelism,
-		clusterEventMap:              clusterEventMap,
+	extenders, err := buildExtenders(options.extenders, options.profiles)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't build extenders: %w", err)
 	}
 
-	metrics.Register()
+	// The nominator will be passed all the way to framework instantiation.
+	nominator := internalqueue.NewPodNominator(informerFactory.Core().V1().Pods().Lister())
+	snapshot := internalcache.NewEmptySnapshot()
+	clusterEventMap := make(map[framework.ClusterEvent]sets.String)
 
-	// Create the config from component config
-	sched, err := configurator.create()
+	profiles, err := profile.NewMap(options.profiles, registry, recorderFactory,
+		frameworkruntime.WithComponentConfigVersion(options.componentConfigVersion),
+		frameworkruntime.WithClientSet(client),
+		frameworkruntime.WithKubeConfig(options.kubeConfig),
+		frameworkruntime.WithInformerFactory(informerFactory),
+		frameworkruntime.WithSnapshotSharedLister(snapshot),
```

Comment:
remove `frameworkruntime.WithRunAllFilters(c.alwaysCheckAllPredicates)` here as `c.alwaysCheckAllPredicates` is always false now. I also create a issue https://github.com/kubernetes/kubernetes/issues/108770 to discuss about whether we should remove this field.

Reply:
yes, we can remove it

Reply:
can you cleanup `WithRunAllFilters` and `runAllFilters` from the framework too

Reply:
This is still be discussed in issue #108770 , if we come to an agreement, I'll create another pr to follow this.